### PR TITLE
[Nano] Fix import path in API docs

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Fix import path in API docs to make it clear for users that how we suggest to import. 

Example change:
- before: although it is not a wrong way here to import, it shows the original path where this function is placed in source code and it is not the recommended way for user
  image
- after:
  image

### Summary of the change 
- [x] Review and (if have problems) fix import path in API docs
   - Nano PyTorch API
     - [x] ~bigdl.nano.pytorch.Trainer~
     - [x] ~bigdl.nano.pytorch.InferenceOptimizer~
     - [x] TorchNano API
     - [x] Patch API
   - Nano Tensorflow API
     - [x] ~bigdl.nano.tf.keras~
     - [x] ~bigdl.nano.tf.optimizers~
     - [x] ~bigdl.nano.tf.keras.InferenceOptimizer~
     - [x] Patch API
   - Nano HPO API
     - [x] ~Search Space~
     - HPO for Tensorflow
       - [x] bigdl.nano.automl.tf.keras.Model
       - [x] bigdl.nano.automl.tf.keras.Sequential
     - HPO for PyTorch
       - [x] ~bigdl.nano.pytorch.Trainer~
- [x] Add api doc for nano_bf16 decorator
- [x] Small fixes to PyTorch Patch API doc strings
- [x] Move api doc for `bigdl.nano.tf.keras.InferenceOptimizer` out of `bigdl.nano.tf.keras` for clearer navigation
- [x] Add api doc for `bigdl.nano.pytorch.patching.patch_encryption`
- [x] Small doc string fixes in  `bigdl.nano.automl.hpo.space.Categorical` and `bigdl.nano.automl.tf.keras.Model/Sequential.search`

### How to test?
- [x] Document test: https://yuwentestdocs.readthedocs.io/en/nano-api-path-fixes/doc/PythonAPI/Nano/index.html

### CODE REVIEW

File: docs/readthedocs/source/doc/PythonAPI/Nano/tensorflow.rst [hunk 1]
```diff
@@ -28,10 +23,18 @@ bigdl.nano.tf.optimizers
     :members: 
     :undoc-members:
 
+bigdl.nano.tf.keras.InferenceOptimizer
+---------------------------------------
+.. autoclass:: bigdl.nano.tf.keras.InferenceOptimizer
+    :members:
+    :undoc-members:
+    :exclude-members: ALL_INFERENCE_ACCELERATION_METHOD
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/PythonAPI/Nano/tensorflow.rst:31]: As we separate the `bigdl.nano.tf.keras.InferenceOptimizer` out of the section `bigdl.nano.tf.keras` section for better navigation purposes, should we also change the title "bigdl.nano.tf.keras" to something else?

**Comment by Oscilloscope98** [comment]
For the HPO Search Space api doc, although in the corresponding [`BigDL/python/nano/src/bigdl/nano/automl/hpo/__init__.py`](https://github.com/intel-analytics/BigDL/blob/main/python/nano/src/bigdl/nano/automl/hpo/__init__.py) file, we have
```python
from .space import *
````
meaning that for the related APIs:
```rst
Search Space
---------------------------

.. autoclass:: bigdl.nano.automl.hpo.space.Categorical

.. autoclass:: bigdl.nano.automl.hpo.space.Real

.. autoclass:: bigdl.nano.automl.hpo.space.Int

.. autoclass:: bigdl.nano.automl.hpo.space.Bool
```
we could import them from `bigdl.nano.automl.hpo` instead of `bigdl.nano.automl.hpo.space` directly. But in the examples, e.g. 
image
they are still imported from `bigdl.nano.automl.hpo.space`.

So, I did not change the import path here.

**Review by MeouSker77 — APPROVED** [review]
LGTM

**Review by rnwang04 — APPROVED** [review]
LGTM

**Pull request merged**